
<!DOCTYPE html>

<html>
  <head>
    <meta charset="utf-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Package Recipe &#39;mosca&#39; &#8212; Bioconda  documentation</title>
    <link rel="stylesheet" href="../../_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="../../_static/alabaster.css" type="text/css" />
    <link rel="stylesheet" type="text/css" href="../../_static/graphviz.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/style.css" />
    <link rel="stylesheet" type="text/css" href="../../_static/font-awesome-4.7.0/css/font-awesome.min.css" />
    <script id="documentation_options" data-url_root="../../" src="../../_static/documentation_options.js"></script>
    <script src="../../_static/jquery.js"></script>
    <script src="../../_static/underscore.js"></script>
    <script src="../../_static/doctools.js"></script>
    <link rel="index" title="Index" href="../../genindex.html" />
    <link rel="search" title="Search" href="../../search.html" />
    <link href="https://fonts.googleapis.com/css?family=Lato|Raleway" rel="stylesheet">
    <link href="https://fonts.googleapis.com/css?family=Inconsolata" rel="stylesheet">

    <link rel="apple-touch-icon" sizes="57x57" href="../../_static/apple-icon-57x57.png">
    <link rel="apple-touch-icon" sizes="60x60" href="../../_static/apple-icon-60x60.png">
    <link rel="apple-touch-icon" sizes="72x72" href="../../_static/apple-icon-72x72.png">
    <link rel="apple-touch-icon" sizes="76x76" href="../../_static/apple-icon-76x76.png">
    <link rel="apple-touch-icon" sizes="114x114" href="../../_static/apple-icon-114x114.png">
    <link rel="apple-touch-icon" sizes="120x120" href="../../_static/apple-icon-120x120.png">
    <link rel="apple-touch-icon" sizes="144x144" href="../../_static/apple-icon-144x144.png">
    <link rel="apple-touch-icon" sizes="152x152" href="../../_static/apple-icon-152x152.png">
    <link rel="apple-touch-icon" sizes="180x180" href="../../_static/apple-icon-180x180.png">
    <link rel="icon" type="image/png" sizes="192x192"  href="../../_static/android-icon-192x192.png">
    <link rel="icon" type="image/png" sizes="32x32" href="../../_static/favicon-32x32.png">
    <link rel="icon" type="image/png" sizes="96x96" href="../../_static/favicon-96x96.png">
    <link rel="icon" type="image/png" sizes="16x16" href="../../_static/favicon-16x16.png">
    <meta name="msapplication-TileColor" content="#ffffff">
    <meta name="msapplication-TileImage" content="../../_static/ms-icon-144x144.png">

    <link rel="manifest" href="/manifest.json">
    <meta name="theme-color" content="#ffffff">

   
  <link rel="stylesheet" href="../../_static/custom.css" type="text/css" />
  
  
  <meta name="viewport" content="width=device-width, initial-scale=0.9, maximum-scale=0.9" />


  </head><body>
  <div class="document">
    
      <div class="sphinxsidebar" role="navigation" aria-label="main navigation">
        <div class="sphinxsidebarwrapper">
<p class="logo">
  <a href="../../index.html">
    <img class="logo" src="../../_static/logo/bioconda_monochrome_small.png" alt="Logo"/>
    
  </a>
</p>








<h3>Navigation</h3>
<ul>
<li class="toctree-l1"><a class="reference internal" href="../../user/index.html">User Docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../contributor/index.html">Contributing to Bioconda</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../developer/index.html">Developer Docs</a></li>
<li class="toctree-l1"><a class="reference internal" href="../../tutorials/index.html">Tutorials</a></li>
</ul>

<ul>
  
  <li class="toctree-l1"><a href="https://github.com/bioconda/bioconda-recipes">Bioconda @ Github</a></li>
  
  <li class="toctree-l1"><a href="../../conda-package_index.html">Package Index</a></li>
  
  <li class="toctree-l1"><a href="https://gitter.im/bioconda/Lobby"><img alt="Gitter" src="https://img.shields.io/gitter/room/bioconda/Lobby.svg"></a></li>
  
</ul>
<div id="searchbox" style="display: none" role="search">
  <h3 id="searchlabel">Quick search</h3>
    <div class="searchformwrapper">
    <form class="search" action="../../search.html" method="get">
      <input type="text" name="q" aria-labelledby="searchlabel" />
      <input type="submit" value="Go" />
    </form>
    </div>
</div>
<script>$('#searchbox').show(0);</script>
        </div>
      </div>
      <div class="documentwrapper">
        <div class="bodywrapper">
          

          <div class="body" role="main">
            
  <div class="section" id="mosca">
<dl class="conda recipe">
<dt>
<em class="property"><span class="pre">recipe</span> </em><code class="sig-name descname"><span class="pre">mosca</span></code></dt>
<dd><p>MOSCA - Meta-Omics Software for Community Analysis</p>
<dl class="field-list simple">
<dt class="field-odd">Homepage</dt>
<dd class="field-odd"><p><a class="reference external" href="https://github.com/iquasere/MOSCA">https://github.com/iquasere/MOSCA</a></p>
</dd>
<dt class="field-even">Documentation</dt>
<dd class="field-even"><p><a class="reference external" href="https://github.com/iquasere/MOSCA/blob/master/README.md">https://github.com/iquasere/MOSCA/blob/master/README.md</a></p>
</dd>
<dt class="field-odd">License</dt>
<dd class="field-odd"><p>GPL / GNU General Public License v3 (GPL-3.0)</p>
</dd>
<dt class="field-even">Recipe</dt>
<dd class="field-even"><p>/<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mosca">mosca</a>/<a class="reference external" href="https://github.com/bioconda/bioconda-recipes/tree/master/recipes/mosca/meta.yaml">meta.yaml</a></p>
</dd>
</dl>
<p>MOSCA (portuguese for fly) is a pipeline designed for performing metagenomics (MG),
metatranscriptomics (MT) and metaproteomics (MP) integrated data analysis,
in a mostly local and fully automated workflow. Metagenomics is used to build
a reference database for MT and MP, beginning with preprocessing of data for
removal of Illumina adapters and lower quality regions, folowed by assembly
of reads into contigs, gene calling on the contigs and homology-based and
domain-based annotation of identified proteins, using the UniProt and COG
databases, respectively. If available, MT reads are then aligned to the ORFs
for gene expression quantification, and MP spectra are submitted for
Peptide-to-Spectrum matching, with the annotated ORFs as reference database.
Final steps include differential expression analysis for both MT and MP, and
metabolic pathways analysis through KEGG Pathways.</p>
</dd></dl>

<dl class="conda package">
<dt id="package-mosca">
<em class="property"><span class="pre">package</span> </em><code class="sig-name descname"><span class="pre">mosca</span></code><a class="headerlink" href="#package-mosca" title="Permalink to this definition">¶</a></dt>
<dd><p><a class="reference external" href="https://anaconda.org/bioconda/mosca"><img alt="(downloads)" src="https://img.shields.io/conda/dn/bioconda/mosca.svg?style=flat" /></a> <a class="reference external" href="https://quay.io/repository/biocontainers/mosca"><img alt="docker_mosca" src="https://quay.io/repository/biocontainers/mosca/status" /></a></p>
<dl class="field-list">
<dt class="field-odd">Versions</dt>
<dd class="field-odd"><details><summary><span class="truncated-version-list"><code>1.3.5-0</code>,  <code>1.3.4-0</code>,  <code>1.3.3-0</code>,  <code>1.3.2-0</code>,  <code>1.3.1-0</code>,  <code>1.3.0-0</code>,  <code>1.2.2-0</code>,  <code>1.2.1-0</code>,  <code>1.2.0-0</code>,  </span></summary><p><code class="docutils literal notranslate"><span class="pre">1.3.5-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.3.4-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.3.3-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.3.2-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.3.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.3.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.2-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.2.0-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.1.1-0</span></code>,  <code class="docutils literal notranslate"><span class="pre">1.1.0-0</span></code></p>
</details></dd>
<dt class="field-even">Depends</dt>
<dd class="field-list-wrapped field-even"><ul class="simple">
<li><p><a class="conda-package reference internal" href="../bioconductor-deseq2/README.html#package-bioconductor-deseq2" title="bioconductor-deseq2 package"><strong>bioconductor-deseq2</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.10.1</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../bioconductor-edger/README.html#package-bioconductor-edger" title="bioconductor-edger package"><strong>bioconductor-edger</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=3.14.0</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../bioconductor-pcamethods/README.html#package-bioconductor-pcamethods" title="bioconductor-pcamethods package"><strong>bioconductor-pcamethods</strong></a> <code class="docutils literal notranslate"><span class="pre">1.82.0.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../bioconductor-rots/README.html#package-bioconductor-rots" title="bioconductor-rots package"><strong>bioconductor-rots</strong></a> <code class="docutils literal notranslate"><span class="pre">1.18.0.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../bioconductor-vsn/README.html#package-bioconductor-vsn" title="bioconductor-vsn package"><strong>bioconductor-vsn</strong></a> <code class="docutils literal notranslate"><span class="pre">3.58.0.*</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/biopython-feedstock"><strong>biopython</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.78</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../blast/README.html#package-blast" title="blast package"><strong>blast</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=2.5.0</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../bowtie2/README.html#package-bowtie2" title="bowtie2 package"><strong>bowtie2</strong></a> <code class="docutils literal notranslate"><span class="pre">2.3.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../checkm-genome/README.html#package-checkm-genome" title="checkm-genome package"><strong>checkm-genome</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.1.0</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../diamond/README.html#package-diamond" title="diamond package"><strong>diamond</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=2.0.2</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../fastqc/README.html#package-fastqc" title="fastqc package"><strong>fastqc</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=0.11.9</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../fraggenescan/README.html#package-fraggenescan" title="fraggenescan package"><strong>fraggenescan</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.31</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../htseq/README.html#package-htseq" title="htseq package"><strong>htseq</strong></a> <code class="docutils literal notranslate"><span class="pre">0.12.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../keggcharter/README.html#package-keggcharter" title="keggcharter package"><strong>keggcharter</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=0.1.1</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../krona/README.html#package-krona" title="krona package"><strong>krona</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=2.7.1</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/lxml-feedstock"><strong>lxml</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=4.6.1</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../maxbin2/README.html#package-maxbin2" title="maxbin2 package"><strong>maxbin2</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=2.2.7</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../maxquant/README.html#package-maxquant" title="maxquant package"><strong>maxquant</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.6.10</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../megahit/README.html#package-megahit" title="megahit package"><strong>megahit</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.2.9</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../metaphlan/README.html#package-metaphlan" title="metaphlan package"><strong>metaphlan</strong></a> <code class="docutils literal notranslate"><span class="pre">3.*</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/openpyxl-feedstock"><strong>openpyxl</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=3.0.3</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/pandas-feedstock"><strong>pandas</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.1.4</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../peptide-shaker/README.html#package-peptide-shaker" title="peptide-shaker package"><strong>peptide-shaker</strong></a> <code class="docutils literal notranslate"><span class="pre">1.16.*</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/perl-feedstock"><strong>perl</strong></a></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/progressbar33-feedstock"><strong>progressbar33</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=2.4</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/python-feedstock"><strong>python</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=3.6</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../quast/README.html#package-quast" title="quast package"><strong>quast</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=5.0.2</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/r-optparse-feedstock"><strong>r-optparse</strong></a></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/r-pheatmap-feedstock"><strong>r-pheatmap</strong></a></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/r-rcolorbrewer-feedstock"><strong>r-rcolorbrewer</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.1.2</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../recognizer/README.html#package-recognizer" title="recognizer package"><strong>recognizer</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.4.0</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/reportlab-feedstock"><strong>reportlab</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=3.5.45</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../samtools/README.html#package-samtools" title="samtools package"><strong>samtools</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.11</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/scikit-learn-feedstock"><strong>scikit-learn</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=0.23.1</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../searchgui/README.html#package-searchgui" title="searchgui package"><strong>searchgui</strong></a> <code class="docutils literal notranslate"><span class="pre">3.3.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../seqkit/README.html#package-seqkit" title="seqkit package"><strong>seqkit</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=0.13.2</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../seqtk/README.html#package-seqtk" title="seqtk package"><strong>seqtk</strong></a></p></li>
<li><p><a class="conda-package reference internal" href="../snakemake/README.html#package-snakemake" title="snakemake package"><strong>snakemake</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=5.27.4</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../sortmerna/README.html#package-sortmerna" title="sortmerna package"><strong>sortmerna</strong></a> <code class="docutils literal notranslate"><span class="pre">2.1b.*</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../spades/README.html#package-spades" title="spades package"><strong>spades</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=3.14.1</span></code></p></li>
<li><p><strong>svn</strong> <code class="docutils literal notranslate"><span class="pre">&gt;=1.9.7</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/tqdm-feedstock"><strong>tqdm</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=4.33.0</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../trimmomatic/README.html#package-trimmomatic" title="trimmomatic package"><strong>trimmomatic</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=0.39</span></code></p></li>
<li><p><a class="conda-package reference internal" href="../upimapi/README.html#package-upimapi" title="upimapi package"><strong>upimapi</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.0.4</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/xlrd-feedstock"><strong>xlrd</strong></a> <code class="docutils literal notranslate"><span class="pre">1.2.*</span></code></p></li>
<li><p><a class="conda-forge reference external" href="https://github.com/conda-forge/xlsxwriter-feedstock"><strong>xlsxwriter</strong></a> <code class="docutils literal notranslate"><span class="pre">&gt;=1.3.7</span></code></p></li>
</ul>
</dd>
<dt class="field-odd">Required By</dt>
<dd class="field-odd"><ul class="simple">
</ul>
</dd>
</dl>
<p class="rubric">Installation</p>
<p>With an activated Bioconda channel (see <a class="reference internal" href="../../user/install.html#set-up-channels"><span class="std std-ref">2. Set up channels</span></a>), install with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">install</span> <span class="n">mosca</span>
</pre></div>
</div>
<p>and update with:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">conda</span> <span class="n">update</span> <span class="n">mosca</span>
</pre></div>
</div>
<p>or use the docker container:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">docker</span> <span class="n">pull</span> <span class="n">quay</span><span class="o">.</span><span class="n">io</span><span class="o">/</span><span class="n">biocontainers</span><span class="o">/</span><span class="n">mosca</span><span class="p">:</span><span class="o">&lt;</span><span class="n">tag</span><span class="o">&gt;</span>
</pre></div>
</div>
<p>(see <a class="reference external" href="https://quay.io/repository/biocontainers/mosca?tab=tags">mosca/tags</a> for valid values for <code class="docutils literal notranslate"><span class="pre">&lt;tag&gt;</span></code>)</p>
</dd></dl>

<div class="section" id="link-to-this-page">
<h2>Link to this page<a class="headerlink" href="#link-to-this-page" title="Permalink to this headline">¶</a></h2>
<p>Render an <a class="reference external" href="http://bioconda.github.io/recipes/mosca/README.html"><img alt="install-with-bioconda" src="https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat" /></a> badge with the following MarkDown:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>[![install with bioconda](https://img.shields.io/badge/install%20with-bioconda-brightgreen.svg?style=flat)](http://bioconda.github.io/recipes/mosca/README.html)
</pre></div>
</div>
</div>
</div>


          </div>
          
        </div>
      </div>
    <div class="clearer"></div>
  </div>
    <div class="footer">
      &copy;2016-2021, The Bioconda Team.
      
      |
      Powered by <a href="http://sphinx-doc.org/">Sphinx 3.5.1</a>
      &amp; <a href="https://github.com/bitprophet/alabaster">Alabaster 0.7.12</a>
      
      |
      <a href="../../_sources/recipes/mosca/README.rst.txt"
          rel="nofollow">Page source</a>
    </div>

    

    <div class="git-ribbon">
      <a href="https://github.com/bioconda/bioconda-recipes/edit/master/recipes/mosca/meta.yaml" rel="me">Edit me on GitHub</a>
    </div>
  </body>
</html>